### DISCUSSION TITLE
Possible to Incorporate 3rd Party Netflow (Elastiflow) Index?

### DISCUSSION THREAD

**Discussion by corycandia** [Security-Onion-Solutions/securityonion#5311] [Unsupported Versions] [answered]
Community,

TL:DR
**Overall Question:** Would it be possible to utalize the SO distributed search node architecture for a 3rd party data set if its being delivered directly into elasticsearch on the manager?  If so, how?

**The long read explanation:**
I've spent the last week or two combing through the SO solution, and I finally think I have a middle school understanding of how data is getting put into logstash, to redis, back to log stash, and into elasticseach.  I don't get excited often, but Security Onion excites me with the multitude and culminating use of individual modern technologies into one beautiful solution.

It does not appear to me that Netflow is a part of the native SO solution even though I see some references to so-flow-* indexes here and there, there's no flow processing modules installed/enabled right?

I am interested in being rebellious and combining network monitoring with network security since the SO architecture is so good.  Elastiflow's output is in the form of sending data directly into ElasticSearch, and automatically creating their index template.  Their template and mappings are good, so I'd like to keep it.  Right now, it's being dropped on the manager, but that's going to: **A** Fill up the manager which will make me do ES ILM, or customize curator, and **B**,  won't take advantage of the scale-able search nodes.

So, my problem is I don't know how to do a custom logstash input for an existing index on the manager, or if that's even possible.  Log stash is just handling the *beats and sending it to redis, so I don't know if I can even do this?  Then, I have to do a custom output for logstash on the search nodes that will throw the data into elastic search / index / template.  

Final questions:
1. Can data already in elastic search be sent to logstash to be forwarded using redis?

2. What would an example of that logstash input config look like?

3. Assuming this was possible, what all needs to be done that I'm missing?
3a. A custom LogStash input defined and applied to manager
3b. A custom output for logstashing sending data into the search node ElasticSeach
3c. Defining the index template and applying it to the search nodes
3d. Creating the kibana index pattern
miss anything?

I know this is a big ask, but if anyone has some thoughts, tips, or examples, I'd be very grateful.  I'm just an infrastructure guy trying to play in a software solution developers world.

I've already gotten some information from this article: #4802 
[Example Elastiflow Document.txt](https://github.com/Security-Onion-Solutions/securityonion/files/7063850/Example.Elastiflow.Document.txt)

**Comment by weslambert** [comment, marked as answer]
Hi Cory,

You might consider the Filebeat Netflow module: 

https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-netflow.html
https://docs.securityonion.net/en/latest/filebeat.html?highlight=filebeat#modules
